VariantShredding.md

```diff
@@ -25,290 +25,316 @@
+| `{ }`                                                                              | null                              | non-null      | null                           | null                                 | null                         | null                               | Object is present but empty                      |
+| null                                                                               | `00` (null)                       | null          |                                |                                      |                              |                                    | Object/value is null                             |
+| missing                                                                            | null                              | null          |                                |                                      |                              |                                    | Object/value is missing                          |
+| INVALID                                                                            | `{"event_type": "login"}`         | non-null      | null                           | `login`                              | null                         | 1729795057774                      | INVALID: Shredded field is present in `value`    |
+| INVALID                                                                            | `"a"`                             | non-null      | null                           | `login`                              | null                         | 1729795057774                      | INVALID: `typed_value` is present for non-object |
+| INVALID                                                                            | `02 00` (object with 0 fields)    | null          |                                |                                      |                              |                                    | INVALID: `typed_value` is null for object        |
 
-This section describes a more deeply nested example, using a top-level array as the shredding type.
+Invalid cases in the table above must not be produced by writers.
+Readers must return an object when `typed_value` is non-null containing the shredded fields.
+If `typed_value` is null and `value` is an object, readers may read the encoded object but are not required to do so.
 
-Below is a sample of JSON that would be fully shredded in this example.
-It contains an array of objects, containing an `a` field shredded as an array, and a `b` field shredded as an integer.
+Readers can assume that a value is not an object if `typed_value` is null and that `typed_value` field values are correct; that is, readers do not need to read the `value` column if `typed_value` fields satisfy the required fields.
```

Comment:
I don't think I understand this, is it possible to illustrate with an example?

Reply:
This is again establishing that the shredded values can always be trusted even if there is an encoding problem.

For example, take the case above where two fields of an object are being projected. This is saying that it is not necessary to read the object's `value` field to check for invalid cases:
* If the shredded fields are defined but `value` does not contain an object
* If the shredded fields are duplicated or conflicting with fields in a `value` object

Reply:
Thanks for the clarification.  I think maybe this should be grouped with line 159 to make the context clearer?

Reply:
Moved.